--- a/ormar/queryset/queryset.py
+++ b/ormar/queryset/queryset.py
@@ -139,13 +139,8 @@
         if func_name in ["sum", "avg"]:
             if any(not x.is_numeric for x in select_actions):
                 raise QueryDefinitionError(
-                    "You can use sum and svg only with numeric types of columns"
+                    "You can use sum and svg only with" "numeric types of columns"
                 )
-        if any(x.field_name not in x.target_model.model_fields for x in select_actions):
-            raise QueryDefinitionError(
-                "You can use aggregate functions only on "
-                "existing columns of the target model"
-            )
         select_columns = [x.apply_func(func, use_label=True) for x in select_actions]
         expr = self.build_select_expression().alias(f"subquery_for_{func_name}")
         expr = sqlalchemy.select(*select_columns).select_from(expr)  # type: ignore
