--- a/nicegui/elements/upload_files.py
+++ b/nicegui/elements/upload_files.py
@@ -3,7 +3,7 @@
 from collections.abc import AsyncIterator
 from dataclasses import dataclass
 from io import BytesIO
-from pathlib import Path, PurePosixPath
+from pathlib import Path
 
 import aiofiles
 import anyio
@@ -149,8 +149,7 @@
         if temp_file:
             await temp_file.close()
 
-    filename = PurePosixPath(upload.filename or '').name  # strips all path components
     if temp_file:
-        return LargeFileUpload(filename, upload.content_type or '', Path(str(temp_file.name)))
+        return LargeFileUpload(upload.filename or '', upload.content_type or '', Path(str(temp_file.name)))
     else:
-        return SmallFileUpload(filename, upload.content_type or '', buffer.getvalue())
+        return SmallFileUpload(upload.filename or '', upload.content_type or '', buffer.getvalue())
