--- a/pycsw/ogc/csw/cql.py
+++ /dev/null
@@ -1,83 +0,0 @@
-import logging
-
-from pycsw.core.etree import etree
-from pycsw.core import util
-from pycsw.ogc.fes.fes1 import MODEL as fes1_model
-
-LOGGER = logging.getLogger(__name__)
-
-
-def cql2fes1(cql, namespaces):
-    """transforms Common Query Language (CQL) query into OGC fes1 syntax"""
-
-    filters = []
-    tmp_list = []
-    logical_op = None
-
-    LOGGER.debug('CQL: %s', cql)
-
-    if ' or ' in cql:
-        logical_op = etree.Element(util.nspath_eval('ogc:Or', namespaces))
-        tmp_list = cql.split(' or ')
-    elif ' OR ' in cql:
-        logical_op = etree.Element(util.nspath_eval('ogc:Or', namespaces))
-        tmp_list = cql.split(' OR ')
-    elif ' and ' in cql:
-        logical_op = etree.Element(util.nspath_eval('ogc:And', namespaces))
-        tmp_list = cql.split(' and ')
-    elif ' AND ' in cql:
-        logical_op = etree.Element(util.nspath_eval('ogc:And', namespaces))
-        tmp_list = cql.split(' AND ')
-
-    if tmp_list:
-        LOGGER.debug('Logical operator found (AND/OR)')
-    else:
-        tmp_list.append(cql)
-
-    for t in tmp_list:
-        filters.append(_parse_condition(t))
-
-    root = etree.Element(util.nspath_eval('ogc:Filter', namespaces))
-
-    if logical_op is not None:
-        root.append(logical_op)
-
-    for flt in filters:
-        condition = etree.Element(util.nspath_eval(flt[0], namespaces))
-
-        etree.SubElement(
-            condition,
-            util.nspath_eval('ogc:PropertyName', namespaces)).text = flt[1]
-
-        etree.SubElement(
-            condition,
-            util.nspath_eval('ogc:Literal', namespaces)).text = flt[2]
-
-        if logical_op is not None:
-            logical_op.append(condition)
-        else:
-            root.append(condition)
-
-    LOGGER.debug('Resulting OGC Filter: %s',
-                 etree.tostring(root, pretty_print=1))
-
-    return root
-
-
-def _parse_condition(condition):
-    """parses a single condition"""
-
-    LOGGER.debug('condition: %s', condition)
-
-    property_name, operator, literal = condition.split()
-
-    literal = literal.replace('"', '').replace('\'', '')
-
-    for k, v in fes1_model['ComparisonOperators'].items():
-        if v['opvalue'] == operator:
-            fes1_predicate = k
-
-    LOGGER.debug('parsed condition: %s %s %s', property_name, fes1_predicate,
-                 literal)
-
-    return (fes1_predicate, property_name, literal)
--- a/pycsw/ogc/csw/csw2.py
+++ b/pycsw/ogc/csw/csw2.py
@@ -6,7 +6,6 @@
 from six.moves.configparser import SafeConfigParser
 from pycsw.core.etree import etree
 from pycsw import oaipmh, opensearch, sru
-from pycsw.ogc.csw.cql import cql2fes1
 from pycsw.plugins.profiles import profile as pprofile
 import pycsw.plugins.outputschemas
 from pycsw.core import config, log, metadata, util
@@ -147,20 +146,12 @@
                     % self.parent.kvp['constraintlanguage'])
                 if self.parent.kvp['constraintlanguage'] == 'CQL_TEXT':
                     tmp = self.parent.kvp['constraint']
-                    try:
-                        LOGGER.debug('Transforming CQL into fes1')
-                        LOGGER.debug('CQL: %s', tmp)
-                        self.parent.kvp['constraint'] = {}
-                        self.parent.kvp['constraint']['type'] = 'filter'
-                        cql = cql2fes1(tmp, self.parent.context.namespaces)
-                        self.parent.kvp['constraint']['where'], self.parent.kvp['constraint']['values'] = fes1.parse(cql,
-                        self.parent.repository.queryables['_all'], self.parent.repository.dbtype,
-                        self.parent.context.namespaces, self.parent.orm, self.parent.language['text'], self.parent.repository.fts)
-                    except Exception as err:
-                        LOGGER.error('Invalid CQL query %s', tmp)
-                        LOGGER.error('Error message: %s', err, exc_info=True)
-                        return self.exceptionreport('InvalidParameterValue',
-                        'constraint', 'Invalid Filter syntax')
+                    self.parent.kvp['constraint'] = {}
+                    self.parent.kvp['constraint']['type'] = 'cql'
+                    self.parent.kvp['constraint']['where'] = \
+                    self.parent._cql_update_queryables_mappings(tmp,
+                    self.parent.repository.queryables['_all'])
+                    self.parent.kvp['constraint']['values'] = {}
                 elif self.parent.kvp['constraintlanguage'] == 'FILTER':
                     # validate filter XML
                     try:
@@ -237,10 +228,8 @@
             maxrecords=self.parent.kvp['maxrecords'],
             startposition=int(self.parent.kvp['startposition'])-1)
         except Exception as err:
-            LOGGER.debug('Invalid query syntax.  Query: %s', self.parent.kvp['constraint'])
-            LOGGER.debug('Invalid query syntax.  Result: %s', err)
             return self.exceptionreport('InvalidParameterValue', 'constraint',
-            'Invalid query syntax')
+            'Invalid query: %s' % err)
 
         dsresults = []
 
