--- a/git/repo/base.py
+++ b/git/repo/base.py
@@ -203,8 +203,6 @@
 
         if not allow_unsafe_protocols:
             Git.check_unsafe_protocols(str(url))
-        if not allow_unsafe_options:
-            Git.check_unsafe_options(options=list(kwargs.keys()), unsafe_options=cls.unsafe_git_clone_options)
         if not allow_unsafe_options and multi_options:
             Git.check_unsafe_options(options=multi_options, unsafe_options=cls.unsafe_git_clone_options)
 
